rust: Support interface description

cathay4t opened the pull request:
Support below configuration:

```yml
---
interfaces:
  - name: eth1
    state: up
    description: testing_nic1
```

Integration test case enabled.

ffmancera commented on rust/src/lib/ifaces/base.rs:
```diff
@@ -218,3 +222,11 @@ fn default_state() -> InterfaceState {
     InterfaceType::Unknown
 }
+
+fn is_option_string_empty(data: &Option<String>) -> bool {
+    if let Some(s) = data {
+        s.is_empty()
+    } else {
+        false
+    }
```
Can data be `None`? Since it is an option I assume yes, so this function will serialize it if so.. is that correct at all? I think the description can be None..

cathay4t replied:
When user ask for `Some("".to_string())`, it means user would like to delete the description. For the seek of verification, we don't show empty description.

ffmancera commented on rust/src/libnm_dbus/connection/user.rs:
```diff
@@ -0,0 +1,61 @@
+// Copyright 2021 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use std::collections::HashMap;
```
Oops wrong header, this code is no licensed under the apache license AFAIK

cathay4t replied:
It is.

cathay4t committed:
rust: Support interface description

Support below configuration:

```yml
---
interfaces:
  - name: eth1
    state: up
    description: testing_nic1
```

Integration test case enabled.

Signed-off-by: Gris Ge <fge@redhat.com>

qinqon commented:
@cathay4t looks like the dependency was merged

cathay4t committed:
test: Remove xfail on `test_ovs_service_missing_with_system_port_only`

The NM 1.36.0 has fixed this problem.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
Ready for review.

ffmancera reviewed (approved):
The description changes looks good to me but I noticed that we cannot remove the xfail.. it is failing..
